enhancements/olm/simplify-olm-apis.md

```diff
@@ -0,0 +1,366 @@
+    version: 2.0.0-alpha
+```
+
+Define, as part of the `Operator` resource, a `status.apis` field that
+
+- contains the `required` and `provided` APIs of an operator
+
+```yaml
+status:
+  apis:
+    provides:
+    - group: how.theydoit.com
+      version: v2alpha1
+      kind: Plumbus
+      plural: plumbai
+      singular: plumbus
+    requires:
+    - group: how.theydoit.com
+      version: v1
+      kind: Grumbo
+      plural: grumbos
+      singular: grumbo
+```
+
+Define, as part of the `Operator` resource, a `status.permissions` field that
+
+- contains the RBAC requirements of an operator
+
+```yaml
+status:
+  cluster: # optional
+  - rules:
+    - apiGroups:
+      - how.theydoit.com
+      resources:
+      - plumbus
+      verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+    serviceAccountName: mr-plumbus
+  namespaced:
+  - rules:
+    - apiGroups:
+      - how.theydoit.com
+      resources:
+      - grumbos
+      verbs:
+      - get
+      - list
+      - watch
+    serviceAccountName: mr-plumbus
+```
+
+Add OLM controller logic, such that for each `Operator` resource
+
+- a single CSV in the operator's component selection is picked
+  - the _newest_ with respect to the `spec.version` field
+- the pick's `displayName`, `description`, and `version` are projected onto the operator's `spec.metadata` field
+- the pick's `required` and `provided` APIs are projected onto the operator's `status.apis` field
+- the pick's `permissions` are projected onto the operator's `status.permissions` field
+
+__Notes:__
+
+Open Questions:
+
+- Why cluster-scoped?
+- Why create a new resource?
+
+#### Aggregate Operator Status
+
+In order to aggregate the status of an operator, the `Operator` resource will:
+
+- surface status conditions for each selected component
+- surface top-level status conditions that describe overall abnormal operator states
+
+__Details:__
+
+Define, as part of the `Operator` resource, a `status.resources.components[*].conditions` field that
+
+- enriches each element with conditions representing the state of the referenced component
+- should be relevant in the context of an operator
+- need not be surfaced for all component kinds
+- should follow [k8s status condition conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+
+```yaml
+status:
+  resources:
+    matchLabels:
+        operators.coreos.com/v2/operators/plumbus: ""
+    components:
+    - kind: ClusterServiceVersion
+      namespace: my-ns
+      name: plumbus.v2.0.0-alpha
+      uid: d70a53b5-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109811
+      conditions:
+      - type: Installing
```

Comment:
I think we should be explicit whether or not `status.resources.components[].conditions[]` are included verbatim or are also "interpreted" by the Operator object. I think they should be verbatim (but perhaps with only certain ones whitelisted/blacklisted?)

Reply:
That's a tough one. I think we want the freedom to both project them directly and create new ones if it makes sense too. We can be more upfront here on what we want to project initially and what to create net new. Would that help?

Reply:
What about:

- `status.resources.conditions[]` are whatever we want, defined in the Operator controller.
- `status.resources.components[].conditions[]` are copied from the underlying resources.

I'm just concerned that it might be confusing to see a condition on the Operator object, then go to the underlying object and see a different condition.